torch/nn/parallel/__init__.py

```diff
@@ -3,7 +3,6 @@
 from .data_parallel import DataParallel, data_parallel
 from .scatter_gather import scatter, gather
 from .distributed import DistributedDataParallel
-from .distributed_cpu import DistributedDataParallelCPU
 
 __all__ = ['replicate', 'scatter', 'parallel_apply', 'gather', 'data_parallel',
-           'DataParallel', 'DistributedDataParallel', 'DistributedDataParallelCPU']
+           'DataParallel', 'DistributedDataParallel']
```

Comment:
We could add a method here:

```python
def DistributedDataParallelCPU(*args, **kwargs):
  # Log warning saying it's getting deprecated.
  return DistributedDataParallel(*args, **kwargs)
```

Then the DDP/CPU users can also benefit from overlapping reduction, if there are still current users.

Otherwise we can postpone merging this until a major version bump.